pkg/bmh/baremetalhost.go

```diff
@@ -457,26 +444,40 @@ func (builder *BmhBuilder) Exists() bool {
 	return err == nil || !k8serrors.IsNotFound(err)
 }
 
-// Get returns bmh object if found.
-func (builder *BmhBuilder) Get() (*bmhv1alpha1.BareMetalHost, error) {
-	if valid, err := builder.validate(); !valid {
-		return nil, err
+// GetBmhOperationalState checks whether the given bmh exists.
+func (builder *BmhBuilder) GetBmhOperationalState() bmhv1alpha1.OperationalStatus {
+	if valid, _ := builder.validate(); !valid {
+		return ""
 	}
 
-	glog.V(100).Infof("Getting baremetalhost %s in namespace %s",
+	glog.V(100).Infof("Pull OperationalStatus value for %s baremetalhost within %s namespace",
 		builder.Definition.Name, builder.Definition.Namespace)
 
-	bmh := &bmhv1alpha1.BareMetalHost{}
-	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
-		Name:      builder.Definition.Name,
-		Namespace: builder.Definition.Namespace,
-	}, bmh)
+	if !builder.Exists() {
+		return ""
+	}
 
-	if err != nil {
-		return nil, err
+	status := builder.Object.Status.OperationalStatus
```

Comment:
This can be simplified
```suggestion
	return builder.Object.Status.OperationalStatus
```